### PR TITLE
Support self-signed ca cert for backends

### LINKED ISSUE #288: Support custom CA cert with backend
https://github.com/restic/restic/pull/1276

### PR DESCRIPTION
**Pull request by Emruz-Hossain**
Fixes #288

### CODE REVIEW

File: pkg/cli/env.go [hunk 1]
```diff
@@ -52,6 +52,8 @@ const (
 	// For authentication based on tokens
 	OS_STORAGE_URL = "OS_STORAGE_URL"
 	OS_AUTH_TOKEN  = "OS_AUTH_TOKEN"
+	//For using certs in Mino server or REST server
```
Comment (diptadas) [inline pkg/cli/env.go]: Fix typo: Mino -> Minio

File: pkg/cli/restic.go [hunk 2]
```diff
@@ -151,3 +211,11 @@ func (w *ResticWrapper) appendCacheDirFlag(args []interface{}) []interface{} {
 	}
 	return append(args, "--no-cache")
 }
+
+func (w *ResticWrapper) getCertFile() (string, error) {
+	certFile := w.scratchDir + "/cacerts/ca.crt"
+	if _, err := os.Stat(certFile); os.IsExist(err) {
```
Comment (diptadas) [inline pkg/cli/restic.go]: shouldn't it be os.IsNotExist() ?

File: pkg/cli/restic.go [hunk 3]
```diff
@@ -151,3 +211,11 @@ func (w *ResticWrapper) appendCacheDirFlag(args []interface{}) []interface{} {
...
+		return "", errors.New(certFile + " not exist.")
```
Comment (diptadas) [inline pkg/cli/restic.go]: wrap original error

File: pkg/cli/restic.go (hunk 2 above)
Comment (Emruz-Hossain) [reply pkg/cli/restic.go]: yes. Fixing.

File: pkg/cli/restic.go [hunk 4]
```diff
@@ -46,14 +50,38 @@ type Snapshot struct {
 func (w *ResticWrapper) ListSnapshots() ([]Snapshot, error) {
 	result := make([]Snapshot, 0)
 	args := w.appendCacheDirFlag([]interface{}{"snapshots", "--json"})
+	if w.cacertEnabled {
```
Comment (diptadas) [inline pkg/cli/restic.go]: add a function w.appendCaCertFlag()

File: pkg/cli/restic.go [hunk 5]
```diff
@@ -151,3 +211,11 @@ func (w *ResticWrapper) appendCacheDirFlag(args []interface{}) []interface{} {
 	}
 	return append(args, "--no-cache")
 }
+
+func (w *ResticWrapper) getCertFile() (string, error) {
+	certFile := w.scratchDir + "/cacerts/ca.crt"
```
Comment (diptadas) [inline pkg/cli/restic.go]: use constant CertFilePath

**Pull request merged**